graph/appender/dead_service.go

```diff
@@ -35,7 +35,7 @@ func applyDeadServices(n *tree.ServiceNode, namespace string, istioClient kubern
 				isDead = true
 			} else {
 				// flag the service if it has a defined service but no pods running for the service version
-				servicePods, err := istioClient.GetServicePods(namespace, serviceName, child.Version, "")
+				servicePods, err := istioClient.GetServicePods(namespace, serviceName, child.Version)
```

Comment:
@jshaughn / @jmazzitelli / @mwringe : in this call I didn't change anything in how pods are related to service, I probably lack some background here. We can get the pods related to the service via `service.Spec.Selector`, is there a reason why we prefer here to build a custom selector our of service name and version?

Reply:
I think we allowed an explicit selector because a service does not have to have a defined selector.  But perhaps we don't need that level of support at this time.

Reply:
Ah, I though services had always a selector.
On the other hand, I guess services can have more complex selectors than just service name & version labels match. Isn't it too restrictive?

Reply:
@jotak can you elaborate on the last point regarding service selector does not include service version? so if there is service A (v1) and service B (v2) - how each one relates to different pods?